### PR TITLE
[SPARK-49078][SQL] Support show columns syntax in v2 table

### PR DESCRIPTION
**Pull request by xuzifu666**
### What changes were proposed in this pull request?

Support v2 table with show columns syntax.

### Why are the changes needed?

In lakehouse format such as Paimon、Iceberg，table are v2 which not support show column syntax，this pr is aimed to support it

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

DataSourceV2SQLSuite

### Was this patch authored or co-authored using generative AI tooling?

no

### CODE REVIEW

**Comment by xuzifu666** [comment]
@zhengruifeng Could you have a review plz?   DataLake table are v2 table，user need support this syntax.

**Comment by xuzifu666** [comment]
@dongjoon-hyun Hi，could you have a review plz？datalake table is v2 table which need this syntax supported.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowColumnsTableExec.scala [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ * Physical plan node for show columns from table.
+ */
+case class ShowColumnsTableExec(
+     output: Seq[Attribute],
+     resolvedTable: ResolvedTable) extends V2CommandExec with LeafExecNode {
+  override protected def run(): Seq[InternalRow] = {
+    val table = resolvedTable.table
+    val rows = new ArrayBuffer[InternalRow]()
+    table.columns().foreach(f => rows += toCatalystRow(f.name()))
+    rows.toSeq
```
Comment (yaooqinn) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowColumnsTableExec.scala:37]: We can simplify this as 

 resolvedTable.table.columns().map(...)...

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowColumnsTableExec.scala [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  }
+
```
Comment (yaooqinn) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowColumnsTableExec.scala:39]: nit: remove blank line

Comment (xuzifu666) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowColumnsTableExec.scala]: Done，Thanks for your review，comments had addressed @yaooqinn

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowColumnsTableExec.scala (hunk 1 above)
Comment (xuzifu666) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowColumnsTableExec.scala]: Done

Comment (yaooqinn) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowColumnsTableExec.scala]: I mean 4 lines into one

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowColumnsTableExec.scala [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+/**
+ * Physical plan node for show columns from table.
+ */
+case class ShowColumnsExec(
+     output: Seq[Attribute],
+     resolvedTable: ResolvedTable) extends V2CommandExec with LeafExecNode {
+  override protected def run(): Seq[InternalRow] = {
+    val rows = new ArrayBuffer[InternalRow]()
+    resolvedTable.table.columns().map(f => rows += toCatalystRow(f.name()))
+    rows.toSeq
```
Comment (yaooqinn) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowColumnsTableExec.scala:36]:
```suggestion
    resolvedTable.table.columns().map(f => toCatalystRow(f.name())).toSeq

```

Comment (yaooqinn) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowColumnsTableExec.scala:36]:
```suggestion
    resolvedTable.table.columns().map(f => toCatalystRow(f.name())).toSeq
```

File: sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala [hunk 4]
```diff
@@ -2377,10 +2377,9 @@ class DataSourceV2SQLSuiteV1Filter
     val t = "testcat.ns1.ns2.tbl"
     withTable(t) {
       spark.sql(s"CREATE TABLE $t (id bigint, data string) USING foo")
-
-      testNotSupportedV2Command("SHOW COLUMNS", s"FROM $t")
-      testNotSupportedV2Command("SHOW COLUMNS", s"IN $t")
-      testNotSupportedV2Command("SHOW COLUMNS", "FROM tbl IN testcat.ns1.ns2")
+      checkAnswer(
+        sql(s"SHOW COLUMNS FROM $t"),
+        Seq(Row("id"), Row("data")))
```
Comment (yaooqinn) [inline sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala:2382]:
```suggestion
      checkAnswer(sql(s"SHOW COLUMNS FROM $t"), Seq(Row("id"), Row("data")))
```

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowColumnsTableExec.scala [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.datasources.v2
+
+import scala.collection.mutable.ArrayBuffer
+
```
Comment (yaooqinn) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowColumnsTableExec.scala:21]:
```suggestion
```

**Comment by yaooqinn** [comment]
Can you retrigger the CI?

**Comment by xuzifu666** [comment]
> Can you retrigger the CI?

OK i try it@yaooqinn

**Review by yaooqinn — APPROVED** [review]
LGTM

**Comment by yaooqinn** [comment]
Merged to master, thank you @xuzifu666

**Comment by cloud-fan** [comment]
late LGTM

**Pull request closed without merging**